### PR TITLE
Clean up dependencies

### PR DESCRIPTION
**Pull request by sellout**

### CODE REVIEW

**Review by wavewave — APPROVED** [review]
Cool! Thanks for this clean-up!

**Comment by zliu41** [comment]
What's the reason for moving version bounds out of `cabal.project`, is it for the benefit of cabal2nix?

**Comment by sellout** [comment]
> What's the reason for moving version bounds out of `cabal.project`, is it for the benefit of cabal2nix?

No, unfortunately it's for the benefit of Hackage (and incidentally cabal2nix and other consumers of cabal files). My (incorrect) assumption was that the stuff in cabal.project would be integrated with the *.cabal files during `cabal sdist`, but this isn't the case. The cabal files are published as-is, without bounds.

**Review by zliu41 — APPROVED** [review]

File: integrations/concat-extensions/integration-test/test/ConCatExtensions/Main.hs [hunk 1]
```diff
@@ -893,10 +892,6 @@ mkTestTerms
         ( const
             [ ( ([t|Pair|], [t|Word8|]),
                 pure ([|Pair <$> Gen.enumBounded <*> Gen.enumBounded|], [|show|])
-              ),
```
[the remark this answers is not in the document]
Comment (sellout) [reply integrations/concat-extensions/integration-test/test/ConCatExtensions/Main.hs:896]: Yeah, basically a remnant of when we had the linear dependency for heavisoft. Which is no longer tied to this code.

**Pull request merged**